src/org/traccar/BaseProtocolDecoder.java

```diff
@@ -31,6 +31,22 @@
 
     private final Protocol protocol;
 
+    public long addUnknownDevice(String uniqueId, SocketAddress remoteAddress) {
+        long deviceId = 0;
+        Device newDevice = new Device();
+        newDevice.setName(uniqueId);
+        newDevice.setUniqueId(uniqueId);
+
+        try {
+            Context.getDeviceManager().addDevice(newDevice);
+            Log.info("Nice meeting new device " + uniqueId + ".");
+        } catch (Exception e) {
```

Comment:
I know that we need catch here, but we don't need to catch `Exception`, we only need to catch `SQLException`. It's a bad practice to catch general exceptions.